### PR TITLE
Place a system metadata stamp on readme chart

### PR DESCRIPTION
**Pull request by scotts**
This also reorients the data file itself to be structured as:
```
{
  "experiments":
    [
      # list of individual experiments
    ]
  "system_metadata":
    {
      # key-value pairs of metadata
    }
}
```
Because this changes the data file, I'll wait until the current experiments are done, and then I'll merge things manually.

### CODE REVIEW

File: benchmarks/decoders/benchmark_decoders_library.py [hunk 1]
```diff
@@ -518,6 +518,14 @@ def plot_data(df_data, plot_path):
         for col in range(video_type_combinations[unique_videos[row]], max_combinations):
             fig.delaxes(axes[row, col])
 
+    plt.gcf().text(
```
Comment (ahmadsharif1) [inline benchmarks/decoders/benchmark_decoders_library.py:521]: I would prefer top left instead of bottom left, but I'll leave it to you

**Review by ahmadsharif1 — APPROVED** [review]

File: benchmarks/decoders/benchmark_decoders_library.py (hunk 1 above)
Comment (ahmadsharif1) [reply benchmarks/decoders/benchmark_decoders_library.py:521]: Also, is_cuda_available isn't that useful. It would be better to get the GPU version

Comment (scotts) [reply benchmarks/decoders/benchmark_decoders_library.py:521]: Done. Honestly, neither look "good". And I realized we need to do either top or bottom; that will be robust to changing the number of videos.

File: benchmarks/decoders/generate_readme_data.py [hunk 2]
```diff
@@ -83,21 +83,24 @@ def main() -> None:
...
-                "cpu_count": os.cpu_count(),
-                "system": platform.system(),
-                "machine": platform.machine(),
-                "python_version": str(platform.python_version()),
-                "is_cuda_available": str(torch.cuda.is_available()),
-            }
-        }
-    )
+    data_for_writing = {
+        "experiments": results,
```
Comment (ahmadsharif1) [inline benchmarks/decoders/generate_readme_data.py:87]: nit: why not call these results?

Comment (scotts) [reply benchmarks/decoders/generate_readme_data.py:87]: When reading a data file, I find it more clear to see "experiments" and "system_metadata". But in the code, I think "results" is more clear that it's the return value of running the experiments.

**Review by ahmadsharif1 — APPROVED** [review]

**Pull request merged**